### PR TITLE
chore: NO-JIRA update component and recipe documentation generation

### PR DESCRIPTION
**Pull request by juliodialpad**
# Update component and recipe documentation generation

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will not increment the version number, but will still deploy to documentation site on release:

- [x] Other (chore)

## :book: Description

- Created `getValidFileList` common function that will be used to extract components and recipes from dialtone to generate documentation and language-server in the future so we don't need to be manually updating `component_list`, `recipes_list`, etc, files.
- Renamed/Moved a few folders and files around as the folders and files should be named the same as the component.
  - `components/tabs` -> `components/tab`
  - `feed_pill` -> `feed_item_pill`
  - `meeting_pill` -> `extensions/meeting_pill`
  - `DtScroller.vue` -> `scroller.vue`
- Updated dialtone-vue documentation generation to include recipes.

## :bulb: Context

- While working on the language server, found that the dialtone-vue documentation was not including the recipes, so I was not able to get that list from anywhere to autocomplete.
- After trying to generate the list of recipes, found that we had a few components which naming was not complying with our standards so my script was not finding them.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [ ] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [ ] I have added / updated unit tests.
- [x] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script. Read docs here: [Dialtone Vue Sync Script](../packages/dialtone-vue3/.github/CONTRIBUTING.md#dialtone-vue-sync-script)
- [ ] I have validated components with a screen reader.
- [ ] I have validated components keyboard navigation.

If new component:

- I am exporting any new components or constants:
  - [ ] from the index.js in the component directory.
  - [x] from the index.js in the root (either `packages/dialtone-vue2` or `packages/dialtone-vue3`).
- [ ] I have added the styles for the new component to the `packages/dialtone-css` package.
- [ ] I have created a page for the new component on the documentation site in `apps/dialtone-documentation`.
- [ ] I have added the new component to `common/components_list.js`
- [ ] I have created a component story in storybook
- [ ] I have created story / stories for any relevant component variants in storybook
- [ ] I have created a docs page for the component in storybook.
- [ ] I have checked that changing all props/slots via the UI in storybook works as expected.

## :crystal_ball: Next Steps

- We'll need to do some investigation and remove/replace the `components_list` file now that we can get it dynamically [PR](https://dialpad.atlassian.net/browse/DLT-2136)

### CODE REVIEW

File: packages/dialtone-vue3/recipes/conversation_view/message_input/message_input.vue [hunk 1]
```diff
@@ -231,7 +231,7 @@ import {
   RICH_TEXT_EDITOR_OUTPUT_FORMATS,
   RICH_TEXT_EDITOR_AUTOFOCUS_TYPES,
 } from '@/components/rich_text_editor';
-import meetingPill from './meeting_pill/meeting_pill';
+import MeetingPill from './extensions/metting_pill/meeting_pill';
```
Comment (braddialpad) [inline packages/dialtone-vue3/recipes/conversation_view/message_input/message_input.vue]: typo, "metting_pill" -> "meeting_pill"

File: packages/dialtone-vue3/components/scroller/scroller.vue [hunk 2]
```diff
@@ -33,6 +33,10 @@ import CoreScroller from './modules/core_scroller.vue';
 import DynamicScroller from './modules/dynamic_scroller.vue';
 import { provide, computed, watch, ref } from 'vue';
 
+defineOptions({
+  name: 'DtScroller',
+});
```
Comment (braddialpad) [inline packages/dialtone-vue3/components/scroller/scroller.vue:38]: Oh nice, don't think there was a way to do this when we first wrote this.

**Review by braddialpad — COMMENTED** [review]
Nice! wanted something like this for a long time.

File: packages/dialtone-vue3/components/scroller/scroller.vue (hunk 2 above)
Comment (juliodialpad) [reply packages/dialtone-vue3/components/scroller/scroller.vue:38]: I think previously you could achieve this by creating another `<script>` tag which is very awful IMO haha, this way is more clear

**Review by braddialpad — APPROVED** [review]

**Pull request merged**